src/main/java/org/isf/menu/gui/MainMenu.java

```diff
@@ -241,28 +242,34 @@ public MainMenu() {
 		// if not internalPharmacies mode remove "medicalsward" menu
 		if (!internalPharmacies) {
 			ArrayList<UserMenuItem> junkMenu = new ArrayList<>();
-			for (UserMenuItem umi : myMenu) {
-				if ("MEDICALSWARD".equalsIgnoreCase(umi.getCode()) || "MEDICALSWARD".equalsIgnoreCase(umi.getMySubmenu())) {
+			Iterator<UserMenuItem> it = myMenu.iterator();
+			while (it.hasNext()) {
+				UserMenuItem umi = it.next();
+				if (umi.getCode().equalsIgnoreCase("MEDICALSWARD") || umi.getMySubmenu().equalsIgnoreCase("MEDICALSWARD"))
 					junkMenu.add(umi);
-				}
 			}
-			for (UserMenuItem umi : junkMenu) {
+			Iterator<UserMenuItem> altIt = junkMenu.iterator();
+			while (altIt.hasNext()) {
+				UserMenuItem umi = altIt.next();
```

Comment:
correct form is with implicit iterator and values as first operand (to avoid NPE)

Reply:
restored current informatici version